datalad/support/network.py

```diff
@@ -342,7 +342,12 @@ def _guess_ri_cls(ri):
 
     if not fields['scheme'] and not fields['hostname']:
         parts = _split_colon(ri)
-        if fields['path'] and '@' in fields['path'] or len(parts) > 1:
+        # if no illegal for hostname/username characters in the first part and
+        if not set(parts[0]).intersection(set('/\\#')) and (
```

Comment:
I was surprised to the `#` given the commit message only mentioned `/` and `\`.

Reply:
hm, indeed... will rewrite